src/commands/xkcd.ts

```diff
@@ -0,0 +1,165 @@
+async function _sendErr(
+	reply: (
+		options:
+			| string
+			| Omit<MessageReplyOptions, 'flags'>
+			| (Omit<InteractionReplyOptions, 'flags'> & {
+					shouldMention?: boolean;
+			  })
+	) => Promise<void>,
+	content: string
+): Promise<void> {
+	await reply({ content: content, ephemeral: true });
+}
+
+export const xkcd: GlobalCommand = {
+	name: 'xkcd',
+	description: 'Fetches the most recent xkcd comic, or a selected ',
+	requiresGuild: false,
+	options: [
+		{
+			name: 'number',
+			description: 'The index number of the comic you would like to have',
+			type: ApplicationCommandOptionType.Integer,
+		},
+	],
+	async execute({ options, reply, sendTyping }) {
+		let comic: string = '';
+		// not making this nullable, instead filling with dummy data to be later filled
+		let results: GetComicResponse = {
+			month: '',
+			num: 0,
+			year: '',
+			title: '',
+			day: '',
+			img: '',
+			link: '',
+			news: '',
+			safe_title: '',
+			alt: '',
+			transcript: '',
+		};
+
+		// let the users know that we are getting a post, and determine which comic to getto get
+		sendTyping();
+		const latestComic = await _latestCheck(); // a  sanity check, getting the most recent comic to know the valid range of comics
+
+		if (latestComic.num === -1) {
+			await _sendErr(reply, 'XKCD call failed. Please try again later.');
+			return;
+		}
+
+		// determinining if a number was passed as an arg
+		const param = options[0];
+		if (param?.value !== undefined) {
+			// number was provided in the command
+			comic = `${param.value as number}`;
+			if (param.value < 1 || param.value > latestComic.num) {
+				// error checking, no negative comics or comic 0. Instead send an error message
+				await _sendErr(
+					reply,
+					`Please insert a valid comic number. The range is 1-${latestComic.num}.`
+				);
+				return;
+			}
+		} else {
+			// no number given, just get the latest
+			comic = 'latest';
+		}
+		if (comic !== 'latest') {
+			// get the comic from the API
+			try {
+				const { data, status } = await axios.get<GetComicResponse>(
+					`https://xkcd.now.sh/?comic=${comic}`
+				);
+				if (status !== 200) {
+					console.log('Axios failed to get a comic');
+					await _sendErr(reply, 'XKCD call returned an error. Please try again later.');
+					return;
+				}
+				results = data;
+			} catch (error) {
+				console.log('Axios failed to get due to exception');
+				console.log(error);
+				await _sendErr(reply, 'XKCD call failed. Please try again later.');
+				return;
+			}
+		} else {
+			// just use the OG call to build the embed
+			results = latestComic;
+		}
+
+		// we should have the data in response, build the embed.
+		const embed = new EmbedBuilder()
+			.setTitle(results.safe_title)
+			.setAuthor({
+				name: `xkcd #${results.num}`,
+			})
+			.setURL(`https://xkcd.com/${results.num}/`)
+			.setImage(results.img)
+			.setDescription(`${results.alt}`)
+			.setColor(0x2b96f3)
```

Comment:
Magic number? What color is this, anyway?

Reply:
Yancey mentioned it as a possible color for use. It's the BYUCS Blue. I'll take it out for now, and if we define a color for embeds in `src/constants/meta` at some point we can add it back in.